### PR TITLE
Menu task force ready to report ✌️ 

### PR DESCRIPTION
**Pull request by TheoPannetier**
View switching is now operational... mostly.

**Fixes:**
- #493 
- #494 
- #495 
- #496 
- #488 
- #489 
- #490 
- #446 
- #499 

This PR raises one issue, however: all windows for `game_view`, `menu_view`, `options_view`... are created at the beginning, regardless of whether the corresponding view is in use or not. This also means that we can only switch once between two views, and not back and forth yet. See #545. Solving this will require rethinking the implementation of the view classes, and I feel this PR is big enough already 😇

### CODE REVIEW

**Comment by richelbilderbeek** [comment]
Ha, I see the problems :-)

Screenshot from 2022-05-18 16-52-10

Whatever needs to be done, the game is still in a runnable state :-)

File: game_view.cpp [hunk 1]
```diff
@@ -1,11 +1,13 @@
 #include "game_view.h"
 
-#ifndef LOGIC_ONLY // that is, compiled on GitHub Actions
+#ifndef LOGIC_ONLY // so this is NOT compiled on GitHub Actions
```
Comment (richelbilderbeek) [inline game_view.cpp:3]: Well spotted!

File: options_view.cpp [hunk 2]
```diff
@@ -1,4 +1,7 @@
 #include "options_view.h"
+#include <cassert>
```
Comment (ollyturner) [inline options_view.cpp:2]: This is perfect. Always use angled brackets for standard header files.

**Review by ollyturner — CHANGES_REQUESTED** [review]
Great Work! Now some people in they really want to can leave the game. Of course they never will :) We only have one very small comment in view_mode.cpp. It was all we could find 👍 After fixing this we definitely accept!

File: options_view.cpp (hunk 2 above)
Comment (TheoPannetier) [reply options_view.cpp:2]: Monkey see, monkey do is most of the time a good policy when editing code 🙈

**Comment by TheoPannetier** [comment]
Ah, can't seem to find the comment in `view_mode.cpp` 😅

**Review by richelbilderbeek — APPROVED** [review]

File: view_mode.cpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#include "view_mode.h"
+#include "cassert"
```
[the remark this answers is not in the document]
Comment (TheoPannetier) [reply view_mode.cpp:2]:
```suggestion
#include <cassert>
```

**Pull request merged**